main/commands/all/wg.go

```diff
@@ -0,0 +1,71 @@
+package all
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/xtls/xray-core/main/commands/base"
+	"golang.org/x/crypto/curve25519"
+)
+
+var cmdWG = &base.Command{
+	UsageLine: `{{.Exec}} wg [-i "private key (base64.StdEncoding)"]`,
+	Short:     `Generate key pair for wireguard key exchange`,
+	Long: `
+Generate key pair for wireguard key exchange.
+
+Random: {{.Exec}} wg
+
+From private key: {{.Exec}} wg -i "private key (base64.StdEncoding)"
+`,
+}
+
+func init() {
+	cmdWG.Run = executeWG // break init loop
+}
+
+var input_curve25519 = cmdWG.Flag.String("i", "", "")
+
+func executeWG(cmd *base.Command, args []string) {
```

Comment:
I think this command is quite useful. But the logic seems identical to x25519? If so, can you try eliminate code duplication (a common function, maybe)